example/src/main/java/com/appunite/rx/example/MainActivity.java

```diff
@@ -48,52 +50,23 @@
     @InjectView(R.id.main_activity_fab)
     FloatingActionButton fab;
 
-    public static class AdapterItemManager implements ViewHolderManager {
-
-        @Override
-        public boolean matches(@Nonnull BaseAdapterItem baseAdapterItem) {
-            return baseAdapterItem instanceof MainPresenter.AdapterItem;
-        }
+    private class MainViewHolder extends SubscriptionViewHolder<MainPresenter.AdapterItem> {
```

Comment:
MainViewHolder can be a static class (it even should be) and instances can be created with dagger via ```Provider<T> ```

btw with java 8 adapter set up is very concise 
```
final UniversalAdapter mainAdapter = new UniversalAdapter(
                ImmutableList.<ViewHolderManager>of(new BaseViewHolderManager<>(R.layout.main_adapter_item, MainViewHolder::new, MainPresenter.AdapterItem.class)));
```